clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -446,4 +463,30 @@ object GarbageCollector {
 
     bulkRemove(df, MaxBulkSize, bucket, region, awsRetries, snPrefix, hcValues).toDF("addresses")
   }
+
+  private def writeParquetReport(
+      dstRoot: String,
+      df: DataFrame,
+      time: String,
+      suffix: String = ""
+  ) = {
+    val dstPath = dstRoot + s"/dt=${time}/${suffix}"
+    df.write.parquet(dstPath)
+  }
+
+  private def writeJsonSummary(
+      dstRoot: String,
+      numDeletedObjects: Long,
+      gcRules: String,
+      hcValues: Broadcast[ConfMap],
+      time: String
+  ) = {
+    val dstPath = new Path(dstRoot + s"/dt=${time}/summary.json")
+    val dstFS = dstPath.getFileSystem(configurationFromValues(hcValues))
+    val jsonSummary = JObject("gc_rules" -> gcRules, "num_deleted_objects" -> numDeletedObjects)
+
+    val stream = dstFS.create(dstPath)
+    stream.writeChars(compact(render(jsonSummary)))
```

Comment:
This leaks `stream` if it throws an exception.  It should catch exceptions and abort the stream on failure.